request/do.go

```diff
@@ -52,21 +52,23 @@ func Do(requests int, quit <-chan struct{}, concurrency int, url string, timeout
 
 	var wg sync.WaitGroup
 
-	for i := 0; i < requests; i++ {
-		select {
-		case <-quit:
-			break
-		default:
+	go func() {
+		for i := 0; i < requests; i++ {
+			select {
+			case <-quit:
+				wg.Wait()
+				close(rec)
+				return
+			default:
+			}
```

Comment:
[Bug] The main problem is fixed, but we still have an issue here.
Given that modification:
```go
for i := 0; i < requests; i++ {
	fmt.Println(i)
	select {
	case <-quit:
		fmt.Println("STOP")
		wg.Wait()
		close(rec)
		return
	default:
	}
	// ...
}
```

and this call:
```sh
> go run cmd/runner/main.go -c 1 -d 1s -r 10 http://localhost:9999\?delay\=500
```

I expect something like:

```txt
0
1
STOP
```

but got instead:

```txt
0
1
2
3
STOP
3
```

So I see 2 problems:
- Too many requests done before STOP
- A duplicated 3 after the STOP

Reply:
Last `3` comes from `println(len(rec))` in main.

For the "too many requests": I get what you mean, but i don't really know what's going on actually 😅

Reply:
> Last `3` comes from `println(len(rec))` in main (number of results). First `3` is printed before the `select` statement.

🤦‍♂️ 

As for the too many request, for some reason the first two seem to pop ± simultaneously. I think we can live with that for now, it'll need to be fixed in the PR for the unit test anyway.

Reply:
yep !